fix(android): Fixed incremental build issues introduced by [TIMOB-27043] in 8.1.0

jquick-axway opened the pull request:
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27135

**Summary:**
- This bug was caught before release.
- Fixed build failure when a local HTML file references a local JS file.
  * Added unit test to exercise this feature on Android and iOS. (Not supported on Windows.)
- Fixed incremental build failure when building for "device" or "production".
  * This is really an encrypted assets issue.

**Test:**
1. Create a Classic app project from default template.
2. Build and run to an Android device. (Not the emulator. Must be a device.)
3. Build and run to the same Android device again.
4. Verify that the build succeeds and runs successfully on device. (This used to fail.)

jquick-axway committed:
[TIMOB-27135] Android: Fixed incremental build issues introduced by [TIMOB-27043] in 8.2.0

- This bug was caught before release.
- Fixed build failure when a local HTML file references a local JS file.
- Fixed incremental build failure when building for "device" or "production".

build commented:
<!--
  0 failure: 
  1 warning:  :mag: Can't find ...
  
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :mag: Can't find junit reports at `./junit.*.xml`, skipping generating JUnit Report.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 674a09b3ebeaef9c3508af2a3dd7fc55b366bd36
</p>

sgtcoolguy committed:
Merge branch 'master' into TIMOB-27135

garymathews reviewed (approved):
CR: **PASS**

jquick-axway committed:
Android: Fixed "forceRebuild" to not attempt an incremental build for [TIMOB-27135]

- Setting the "forceRebuild" flag now deletes the new "incremental" build directory.

garymathews committed:
Merge branch 'master' into TIMOB-27135

jquick-axway committed:
Android: Modified build system to rebuild app if "transpile" or "source-maps" settings changed

keerthi1032 commented:
FR passed. No error shown and app installed and launched successfully.
**Test Environment:**
Name                        = Mac OS X
  Version                     = 10.13.6
  Architecture                = 64bit
Node.js
  Node.js Version             = 10.13.0
  npm Version                 = 6.4.1
Titanium CLI
  CLI Version                 = 5.1.1
Titanium SDK
  SDK Version                 = local sdk 8.2.0.v20190611120304
CLI =7.0.11
Device =Oneplus 5T android 9,Samsung s5 android 6 and nexus android 5
Emulator =Pixel xl android 9

keerthi1032 committed:
Merge branch 'master' into TIMOB-27135

garymathews committed:
Merge branch 'master' into TIMOB-27135
